gtk4/src/subclass/widget.rs

```diff
@@ -588,7 +588,7 @@ unsafe impl<T: WidgetImpl> IsSubclassable<T> for Widget {
             let mut data = T::type_data();
             let data = data.as_mut();
             // Used to store actions for `install_action`
-            data.set_class_data(Widget::static_type(), Actions::default());
+            data.set_class_data(<T as ObjectSubclassType>::type_(), Actions::default());
```

Comment:
That's also suboptimal. The class data is owned by the class, not your code :)

If you need to store multiple things in there you'll have to store a `HashMap` or so.

Reply:
Actually it doesn't seem to be the issue at all and this change isn't relevant anymore. The code in Fractal had the wrong class_init/instance_init. It wasn't chaining up the right parent and I didn't notice it till later. Although I thought this change wasn't very probelmatic at the end , or is it ?

Reply:
Yes, it would be a problem if the subclass wants to store class data itself. You could conflict then.

Reply:
Ah I see. I had a huge mis-undertanding how it all worked and now it's much clearer :) Thanks!